adaptation.lua

```diff
@@ -0,0 +1,110 @@
+-- Original opening comment before lua adaptation
+-- View or set cavern adaptation levels
+-- based on removebadthoughts.rb
+-- Rewritten by TBSTeun using OpenAI GPT from adaptation.rb
+
+local function print_color(color, s)
+    dfhack.color(color)
+    dfhack.print(s)
+    dfhack.color(COLOR_RESET)
+end
+
+local function usage(s)
+    if s then
+        print(s)
+    end
+    print(dfhack.script_help())
+end
+
+local function set_adaptation_value(unit, v)
+    local num_set = 0
+
+    if not dfhack.units.isCitizen(unit) or unit.flags2.killed then
+        return num_set
+    end
+    local trait_found = false
+    for _, t in ipairs(unit.status.misc_traits) do
+        if t.id == df.misc_trait_type.CaveAdapt then
+            if mode == "show" then
+                    print_color(COLOR_RESET, "Unit " .. unit.id .. " (" .. dfhack.TranslateName(dfhack.units.getVisibleName(unit)) .. ") has an adaptation of ")
+                    if t.value <= 399999 then
+                        print_color(COLOR_GREEN, t.value .. "\n")
+                    elseif t.value <= 599999 then
+                        print_color(COLOR_YELLOW, t.value .. "\n")
+                    else
+                        print_color(COLOR_RED, t.value .. "\n")
+                    end
+            elseif mode == "set" then
+                print("Unit " .. unit.id .. " (" .. dfhack.TranslateName(dfhack.units.getVisibleName(unit)).. ") changed from " .. t.value .. " to " .. v)
+                t.value = v
+                num_set = num_set + 1
+            end
+            trait_found = true
+            break
+        end
+    end
+    if not trait_found then
+        if mode == "show" then
+            print_color(COLOR_RESET, "Unit " .. unit.id .. " (" .. dfhack.TranslateName(dfhack.units.getVisibleName(unit)) .. ") has an adaptation of ")
+            print_color(COLOR_GREEN, "0\n")
+        elseif mode == "set" then
+            local new_trait = df.unit_misc_trait:new()
+            new_trait.id = df.misc_trait_type.CaveAdapt
+            new_trait.value = v
+            num_set = num_set + 1
+            table.insert(unit.status.misc_traits, new_trait)
+            print("Unit " .. unit.id .. " (" .. dfhack.TranslateName(dfhack.units.getVisibleName(unit)) .. ") changed from 0 to " .. v)
+        end
+    end
+
+    return num_set
+end
+
+local args = {...}
+
+local mode = args[1] or "help"
+local who = args[2]
+local value = args[3]
+
+if mode == "help" then
+    usage()
```

Comment:
you might want to `return` after every call to `usage()` here and below. Otherwise, you might incorrectly run processing code on bad input. For example, a command like `adaptation help him please` will print out help text and then throw an error when 'please' is used as a number.